stacks-signer/src/client/stacks_client.rs

```diff
@@ -58,35 +66,95 @@ pub struct StacksClient {
     tx_version: TransactionVersion,
     /// The chain we are interacting with
     chain_id: u32,
+    /// Whether we are mainnet or not
+    mainnet: bool,
     /// The Client used to make HTTP connects
     stacks_node_client: reqwest::blocking::Client,
 }
 
-/// The supported epoch IDs
-#[derive(Debug, PartialEq)]
-pub enum EpochId {
-    /// The mainnet epoch ID
-    Epoch30,
-    /// The testnet epoch ID
-    Epoch25,
-    /// Unsuporrted epoch ID
-    UnsupportedEpoch,
-}
-
-impl From<&Config> for StacksClient {
-    fn from(config: &Config) -> Self {
+impl From<&GlobalConfig> for StacksClient {
+    fn from(config: &GlobalConfig) -> Self {
         Self {
             stacks_private_key: config.stacks_private_key,
             stacks_address: config.stacks_address,
             http_origin: format!("http://{}", config.node_host),
             tx_version: config.network.to_transaction_version(),
             chain_id: config.network.to_chain_id(),
             stacks_node_client: reqwest::blocking::Client::new(),
+            mainnet: config.network.is_mainnet(),
         }
     }
 }
 
 impl StacksClient {
+    /// Create a new signer StacksClient with the provided private key, stacks node host endpoint, and version
+    pub fn new(stacks_private_key: StacksPrivateKey, node_host: SocketAddr, mainnet: bool) -> Self {
+        let pubkey = StacksPublicKey::from_private(&stacks_private_key);
+        let tx_version = if mainnet {
+            TransactionVersion::Mainnet
+        } else {
+            TransactionVersion::Testnet
+        };
+        let chain_id = if mainnet {
+            CHAIN_ID_MAINNET
+        } else {
+            CHAIN_ID_TESTNET
+        };
+        let stacks_address = StacksAddress::p2pkh(mainnet, &pubkey);
+        Self {
+            stacks_private_key,
+            stacks_address,
+            http_origin: format!("http://{}", node_host),
+            tx_version,
+            chain_id,
+            stacks_node_client: reqwest::blocking::Client::new(),
+            mainnet,
+        }
+    }
+
+    /// Get our signer address
+    pub fn get_signer_address(&self) -> &StacksAddress {
+        &self.stacks_address
+    }
+
+    /// Calculate the ordered list of coordinator ids by comparing the provided public keys against the pox consensus hash
+    pub fn calculate_coordinator_ids(&self, public_keys: &PublicKeys) -> Vec<u32> {
+        let pox_consensus_hash = match retry_with_exponential_backoff(|| {
```

Comment:
Do we want to call this with retry again? Since get_pox_consensus_hash already makes a retry call to get RPCPeerInfoData

Reply:
Def not. Will remove :)